src/library/tactic/list_axioms_tactic.h

```diff
@@ -0,0 +1,34 @@
+/*
+Copyright (c) 2017 Robert Y. Lewis. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+
+Author: Robert Y. Lewis
+*/
+
+#pragma once
+#include "util/name_set.h"
+
+namespace lean {
+
+struct list_axioms_deps {
+    environment     m_env;
+    name_set        m_visited;
+    name_set        m_axioms;
+    bool            m_use_axioms;
+    bool            m_used_sorry;
+    list_axioms_deps(environment const & env):
+        m_env(env), m_use_axioms(false), m_used_sorry(false) {}
+
+    void visit(name const & n);
+    void visit(expr const & e);
+    buffer<name> axiom_list();
+    buffer<name> operator()(name const & n) {

```

Comment:
Yes, this was only a slight modification of the `#print axioms` code at https://github.com/leanprover/lean/blob/d610709a5358de741f5094ccdfed6241434252e2/src/frontends/lean/print_cmd.cpp#L39 . They should share one function, I just wasn't sure where that function should live. It doesn't seem right for `print_cmd.cpp` to import something in the `tactic` folder, no?